src/pl/michal/interfejs/MySQLaccess.java

```diff
@@ -0,0 +1,202 @@
+            try {
+                if (connection != null)
+                    connection.rollback();
+            } catch (Exception ex) {
+                throw ex;
+            }
+            throw e;
+        }
+    }
+
+    public void FinishTransaction() throws Exception {
+        try {
+            if (connection != null)
+                connection.commit();
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    public void CloseDB() throws Exception {
+        try {
+            if (resultSet != null)
+                resultSet.close();
+            if (preparedStatement != null)
+                preparedStatement.close();
+            if (statement != null)
+                statement.close();
+            if (connection != null)
+                connection.close();
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    public double GetAccountInfo(int accountNO) throws Exception {
+        if (accountNO <= 0) {
+            throw new IllegalArgumentException("No passed Account number");
+        }
+
+        try {
+            statement = connection.createStatement();
+//            resultSet=statement.executeQuery(String.format("SELECT accountBalance FROM accountData WHERE accountNO=%d",accountNO));
+            resultSet = statement.executeQuery("SELECT accountBalance FROM accountData WHERE accountNO=" + accountNO);
+            if (resultSet.next()) {
+                return resultSet.getDouble(1);
+            }
+            return - 1;
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    public void UpdateAccountData(int accountNO, double val, int transferID) throws Exception {
+        if (accountNO <= 0 || val < 0 || transferID <= 0) {
+            throw new IllegalArgumentException("Wrong argument data\n");
+        }
+        String sqlUpdate = "UPDATE accountData "
+                + "SET accountBalance = ?,"
+                + "bankTransaction_transactionID = ? "
+                + "WHERE accountNO = ?";
+        try {
+            preparedStatement = connection.prepareStatement(sqlUpdate);
+            preparedStatement.setDouble(1, val);
+            preparedStatement.setInt(2, transferID);
+            preparedStatement.setInt(3, accountNO);
+            preparedStatement.executeUpdate();
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    public void InsertBankTransaction(int from, int to, double amount, int fee) throws Exception {
+        if (from <= 0 || to <= 0 || amount <= 0) {
+            throw new IllegalArgumentException("Wrong argument data\n");
+        }
+        //Columns:
+        //transactionID int(10) UN AI PK
+        //accountFrom int(11)
+        //accountTo int(11)
+        //amount double
+        //transactionDate timestamp
+        try {
+            preparedStatement = connection.prepareStatement("INSERT INTO bankTransaction VALUES(default,?,?,?,?)", Statement.RETURN_GENERATED_KEYS);
```

Comment:
Bardzo dobrze, żę używasz 'prepared statement', bo można też sklejać operacje SQL stringami, ale wtedy to rodzi ryzyka, których unikasz w ten sposób.